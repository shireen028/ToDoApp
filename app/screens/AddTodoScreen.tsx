import React, {useState} from 'react';
import {View, StyleSheet, Alert, ScrollView} from 'react-native';
import {Button, Text} from 'react-native-elements';
import {Input} from 'components/Input';
import {Navigation} from 'react-native-navigation';

type Props = {
  componentId: string;
  addTodo: Function;
};

export const AddTodoScreen: React.FC<Props> = ({componentId, addTodo}) => {
  const [text, setText] = useState('');
  const [loading, setLoading] = useState(false);

  const handleAddTodo = async () => {
    setLoading(true);
    try {
      const response = await fetch(
        'https://jsonplaceholder.typicode.com/todos',
        {
          headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json',
            'Cache-Control': 'no-cache',
          },
          method: 'POST',
          body: JSON.stringify({title: text, userId: 1, completed: false}),
        },
      );
      const todo = await response.json();
      addTodo(todo);
    } catch (error) {
      Alert.alert('Error', 'Create todo failed');
    }
    setLoading(false);
    Navigation.pop(componentId);
  };
  return (
    <ScrollView>
      <View style={styles.container}>
        <View style={styles.content}>
          <Text h3 style={styles.title}>
            Create a Todo
          </Text>
          <View style={{alignItems: 'center', marginVertical: 40}}>
            <Input
              placeholder="Todo"
              value={text}
              onChangeText={setText}
              multiline
              onSubmitEditing={handleAddTodo}
            />
            <Button
              title="Add Todo"
              onPress={handleAddTodo}
              loading={loading}
              icon={styles.icon}
              buttonStyle={styles.button}
            />
          </View>
        </View>
      </View>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    padding: 20,
  },
  content: {
    backgroundColor: 'white',
    borderRadius: 10,
    minHeight: '60%',
  },
  title: {
    textAlign: 'center',
    marginVertical: 20,
  },
  button: {
    minHeight: 52,
    minWidth: '90%',
    backgroundColor: 'blue',
    marginVertical: 50,
  },
  icon: {
    name: 'add-outline',
    type: 'ionicon',
    color: 'white',
  },
});
