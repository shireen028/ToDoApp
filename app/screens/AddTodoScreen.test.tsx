import 'react-native';
import React from 'react';
import {AddTodoScreen} from './AddTodoScreen';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<AddTodoScreen componentId="sdfs" addTodo={() => {}} />);
  expect(tree).toMatchSnapshot();
});
