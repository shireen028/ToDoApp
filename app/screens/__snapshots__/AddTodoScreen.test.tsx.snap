// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "padding": 20,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 10,
            "minHeight": "60%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#242424",
              "fontSize": 36,
              "marginVertical": 20,
              "textAlign": "center",
            }
          }
        >
          Create a Todo
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "marginVertical": 40,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "height": 100,
                  "justifyContent": "space-between",
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginVertical": 12,
                  "paddingHorizontal": 20,
                  "paddingVertical": 15,
                },
                Object {
                  "borderColor": "#F2EBDD",
                },
              ]
            }
          >
            <TextInput
              multiline={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Todo"
              style={
                Object {
                  "color": "black",
                  "fontSize": 16,
                  "minHeight": 100,
                  "width": "98%",
                }
              }
              value=""
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "borderRadius": 3,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 3,
                },
                undefined,
                false,
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "busy": false,
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "blue",
                    "borderColor": "#2089dc",
                    "borderRadius": 3,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginVertical": 50,
                    "minHeight": 52,
                    "minWidth": "90%",
                    "padding": 8,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "marginHorizontal": 5,
                      "overflow": "hidden",
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "white",
                              "fontSize": 24,
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                            Object {
                              "fontFamily": "Ionicons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                        testID="iconIcon"
                      >
                        ï„„
                      </Text>
                    </View>
                  </View>
                </View>
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 18,
                      "paddingVertical": 1,
                      "textAlign": "center",
                    }
                  }
                >
                  Add Todo
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
